packages/x-date-pickers/src/DateTimePicker/DateTimePicker.tsx

```diff
@@ -278,7 +279,7 @@ DateTimePicker.propTypes = {
    * The date used to generate the new value when both `value` and `defaultValue` are empty.
    * @default The closest valid date-time using the validation props, except callbacks like `shouldDisable<...>`.
    */
-  referenceDate: PropTypes.any,
+  referenceDate: PropTypes.oneOfType([PropTypes.instanceOf(Date), PropTypes.object]),
```

Comment:
OK this is the major drawback of the new approach..
The proptype generation script is smart with `Date` object does not just mark them as "object", so we end up with a proptype that is wrong for people not using date-fns...

Reply:
I could add a param in `getPropTypesFromFile` `ignoreDateType` or something similar, it's not great but I think it's worth it here

Reply:
Yes, this case would need some extra attention in order to avoid this particular change. 🙈

Reply:
Fixed by https://github.com/mui/material-ui/pull/40774